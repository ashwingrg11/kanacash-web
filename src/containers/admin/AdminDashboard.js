import React, { Component } from "react";
import PropTypes from "prop-types";
import { Row, Col, Card } from "react-bootstrap";
import COLORS from "../../assets/Color/Color";
import FeeSetIcon from "../../assets/images/FeeSetIcon.svg";
import UnlockSenderImg from "../../assets/images/admin/unlockSenderImg.svg";
import CustomTable from "../../components/Table/CustomTable";
import { getActiveAccountSideBar } from "../../store/actions/senderDetail";

// import * as api from "../../services/axios/adminApi";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { isEmpty } from "../../utils/isEmpty";
import ROUTES from "../../assets/Routes/Routes";
class AdminDashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tabelData: [],
      tableHeader: [
        "PAYMENT METHOD",
        "PAYOUT METHOD",
        "CURRENCY",
        "MIN AMOUNT",
        "MAX AMOUNT",
        "FLAT FEE",
        "% FEE",
        "Action",
      ],
    };
  }

  componentDidMount = () => {
    if (!isEmpty(this.props.feeStructure))
      this.setState({ tabelData: this.props.feeStructure });
  };

  componentDidUpdate = (prevProps) => {
    if (prevProps.feeStructure !== this.props.feeStructure) {
      this.setState({ tabelData: this.props.feeStructure });
    }
  };
  render() {
    return (
      <>
        <Row>
          <Col
            style={{
              color: `${COLORS.BLACK_COLOR}`,
              fontWeight: "bold",
              fontSize: "22px",
            }}
          >
            Dashboard
          </Col>
        </Row>

        <Row className="mt-3">
          <Col lg="6 mb-3 mb-lg-0">
            <div
              className="blue-bg"
              onClick={() => {
                this.props.getActiveAccountSideBar("AddFeeSet");
                localStorage.setItem("activeSideBar", "AddFeeSet");

                this.props.history.push(`${ROUTES.ADD_FEE_SET}`);
              }}
            >
              <img src={FeeSetIcon} alt="FeeSetIcon" />
              <h3>Add Fee Set</h3>
            </div>
          </Col>
          <Col lg="6 mb-1 mb-lg-0">
            <div
              className="blue-bg"
              onClick={() => {
                this.props.getActiveAccountSideBar("UnlockSender");
                localStorage.setItem("activeSideBar", "UnlockSender");

                this.props.history.push(`${ROUTES.UNLOCK_SENDER}`);
              }}
            >
              <img src={UnlockSenderImg} alt="UnlockSenderImg" />
              <h3> Unlock Senders </h3>
            </div>
          </Col>{" "}
        </Row>
        <Row className="mt-4">
          <Col className="subhead-title">Fee Set</Col>
          <Col className="d-flex justify-content-end ">
            {" "}
            {/* <div className="btn-link">View all</div> */}
          </Col>
        </Row>
        <Col className="mt-2 p-0">
          <Card className="p-3">
            <div className="table-responsive">
              <CustomTable
                dataTableData={this.state.tabelData}
                tableHeader={this.state.tableHeader}
                keys={this.state.keys}
                feeRangesKey={this.state.feeRangesKey}
              ></CustomTable>
            </div>
          </Card>
        </Col>
      </>
    );
  }
}

AdminDashboard.propTypes = {
  feeStructure: PropTypes.array,
  history: PropTypes.object,
  getActiveAccountSideBar: PropTypes.func,
};

const mapStateToProps = (state) => ({
  feeStructure: state.admin.feeStructure,
});

const mapDispatchToProps = { getActiveAccountSideBar };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(AdminDashboard));
