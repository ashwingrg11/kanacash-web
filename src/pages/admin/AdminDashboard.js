import React, { Component } from "react";
import Header from "../../containers/layout/Header";
import { Container, Row, Col } from "react-bootstrap";
import Footer from "../../containers/layout/Footer";
import AdminSideBar from "../../containers/admin/AdminSideBar";
import Dashboard from "../../containers/admin/AdminDashboard";
import Disclaimer from "../../components/Disclaimer/Disclaimer";
import { connect } from "react-redux";
class AdminDashboard extends Component {
  render() {
    return (
      <>
        <Header />
        <div className="shadow-line"></div>
        <Container>
          <Row
            style={{
              minHeight: "900px",
            }}
          >
            <Col>
              <Row>
                <Col xl="3" lg="4" md="5" sm="12" style={{ marginTop: "40px" }}>
                  <AdminSideBar />
                </Col>
                <Col xl="9" lg="8" md="7" sm="12" style={{ marginTop: "40px" }}>
                  <Dashboard />
                </Col>
              </Row>
            </Col>
          </Row>
        </Container>

        <Footer />
        {!localStorage.acceptPolicy && <Disclaimer />}
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  acceptPolicy: state.auth.accept_policy,
});

export default connect(mapStateToProps)(AdminDashboard);
