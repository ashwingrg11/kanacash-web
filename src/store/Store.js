import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
