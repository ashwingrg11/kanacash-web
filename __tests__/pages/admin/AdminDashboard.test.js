import React from "react";
import AdminDashboard from "../../../src/pages/admin/AdminDashboard";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import store from "../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<AdminDashboard />", () => {
  it("AdminDashboard renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AdminDashboard />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
