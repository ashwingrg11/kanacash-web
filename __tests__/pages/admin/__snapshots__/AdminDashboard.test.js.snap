// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminDashboard /> AdminDashboard renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className="shadow-line"
  />,
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "minHeight": "900px",
        }
      }
    >
      <div
        className="col"
      >
        <div
          className="row"
        >
          <div
            className="col-xl-3 col-lg-4 col-md-5 col-sm-12"
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <div
              className="row"
              style={
                Object {
                  "backgroundColor": "#FAFAFA",
                  "padding": "20px",
                }
              }
            >
              <div
                className="col"
              >
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Dashboard
                  </div>
                </div>
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Add Fee Set
                  </div>
                </div>
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Fee Set
                  </div>
                </div>
                <div
                  className="mt-2 row"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <div
                    className="d-flex align-items-center col"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Unlock Sender
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-xl-9 col-lg-8 col-md-7 col-sm-12"
            style={
              Object {
                "marginTop": "40px",
              }
            }
          >
            <div
              className="row"
            >
              <div
                className="col"
                style={
                  Object {
                    "color": "#212529",
                    "fontSize": "22px",
                    "fontWeight": "bold",
                  }
                }
              >
                Dashboard
              </div>
            </div>
            <div
              className="mt-3 row"
            >
              <div
                className="col-lg-6 mb-3 mb-lg-0"
              >
                <div
                  className="blue-bg"
                  onClick={[Function]}
                >
                  <img
                    alt="FeeSetIcon"
                    src=""
                  />
                  <h3>
                    Add Fee Set
                  </h3>
                </div>
              </div>
              <div
                className="col-lg-6 mb-1 mb-lg-0"
              >
                <div
                  className="blue-bg"
                  onClick={[Function]}
                >
                  <img
                    alt="UnlockSenderImg"
                    src=""
                  />
                  <h3>
                     Unlock Senders 
                  </h3>
                </div>
              </div>
               
            </div>
            <div
              className="mt-4 row"
            >
              <div
                className="subhead-title col"
              >
                Fee Set
              </div>
              <div
                className="d-flex justify-content-end  col"
              >
                 
              </div>
            </div>
            <div
              className="mt-2 p-0 col"
            >
              <div
                className="p-3 card"
              >
                <div
                  className="table-responsive"
                >
                  <table
                    id="customTable"
                  >
                    <tbody>
                      <tr>
                        <th>
                          PAYMENT METHOD
                        </th>
                        <th>
                          PAYOUT METHOD
                        </th>
                        <th>
                          CURRENCY
                        </th>
                        <th>
                          MIN AMOUNT
                        </th>
                        <th>
                          MAX AMOUNT
                        </th>
                        <th>
                          FLAT FEE
                        </th>
                        <th>
                          % FEE
                        </th>
                        <th>
                          ACTION
                        </th>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
